src/pt/leitordemanga/src/eu/kanade/tachiyomi/extension/pt/leitordemanga/LeitorDeManga.kt

```diff
@@ -17,5 +24,50 @@ class LeitorDeManga : Madara(
 
     override val client: OkHttpClient = super.client.newBuilder()
         .rateLimit(1, 2, TimeUnit.SECONDS)
+        .addInterceptor(::jsChallengeInterceptor)
         .build()
+
+    // Linked to src/en/yakshascans
+    private fun jsChallengeInterceptor(chain: Interceptor.Chain): Response {
+        val response = chain.proceed(chain.request())
+        if (response.code != 403) {
+            return response
+        }
+        response.close()
+
+        Thread.sleep(3000L)
+        val token = fetchToken(chain).sha256()
+        val body = FormBody.Builder()
+            .add("challenge", token)
+            .build()
+
+        chain.proceed(POST("$baseUrl/hcdn-cgi/jschallenge-validate", headers, body)).use {
+            if (it.isSuccessful.not()) {
+                throw IOException("Failed to bypass js challenge!")
+            }
+        }
+        return chain.proceed(chain.request())
+    }
+
+    private fun fetchToken(chain: Interceptor.Chain, attempt: Int = 0): String {
+        if (attempt > MAX_ATTEMPT) {
+            throw IOException("Failed to fetch challenge token!")
+        }
+        return chain.proceed(GET("$baseUrl/hcdn-cgi/jschallenge", headers)).let {
+            TOKEN_REGEX.find(it.body.string())?.groups?.get(1)?.value
+                ?.takeUnless { it == "nil" } ?: fetchToken(chain, attempt.plus(1))
+        }
+    }
```

Comment:
Rewrite like this to make it `tailrec`.

```suggestion
    private tailrec fun fetchToken(chain: Interceptor.Chain, attempt: Int = 0): String {
        if (attempt > MAX_ATTEMPT) {
            throw IOException("Failed to fetch challenge token!")
        }
        val response = chain.proceed(GET("$baseUrl/hcdn-cgi/jschallenge", headers)).body.string()
        val token = TOKEN_REGEX.find(response)?.run { groups[1]!!.value }?.takeUnless { it == "nil" }
        return token ?: fetchToken(chain, attempt + 1)
    }
```